csharp/source/autowrap/csharp/boilerplate.d

```diff
@@ -102,11 +104,45 @@ private string commonBoilerplate() @safe pure {
             }
         }
 
+        mixin(generateSharedTypes());
```

Comment:
A mixin in a mixin? Won't that make it hard to understand what's going on?

Reply:
In general yes, however, in this case I think it should be OK as all we are doing is mixing in one (maybe more later) type, and not code that needs to be debugged.

Reply:
If this mixin is changed then someone might have to debug it if it breaks. Isn't there a simpler way to do this?

Reply:
Fixed.